src/main/resources/avro/references.avdl

```diff
@@ -1,24 +1,144 @@
 @namespace("org.ga4gh")
 
+/**
+We represent reference genomes, or collections therof, by sequence graphs.
+
+A sequence graph is made by progressively joining new sequence pieces, which we
+call `Reference`s, into an existing graph.  This starts with a primary sequence.
+For example consider the primary sequence in GRCh38 for a chromosome. Then
+additional `Reference`s can be joined into it, starting on a defined side of a
+defined base in the existing structure, containing new sequence (potentially
+empty if a pure deletion or breakpoint) and then joining back to another defined
+side of another base in the current graph.
+
+Any base in the graph will have a unique identity defined by its `Reference` id
+and 0-based position in the `Reference` sequence. It is oriented in the
+direction of the `Reference` sequence.
+
+We support a context model for uniquely identifying bases in the graph.
+Currently this is the UCSC left-right mapping proposal as proposed by Paten,
+Haussler et al. - http://arxiv.org/abs/1404.5010
+*/
 protocol References {
 
 /**
+Each `Segment` in a sequence graph has two sides, PLUS and MINUS.
+
+`PLUS` represents forwards, or the direction of increasing coordinates, while
+`MINUS` represents reverse-complement, or the direction of decreasing
+coordinates.
+*/
+enum SegmentSide {
+  PLUS,
+  MINUS
+}
+
+
+/**
+`Segment`s are joined into a sequence graph by being connected at their ends
+to positions along other `Segment`s.
+*/
+record SegmentJoinLocation {
+  /**
+  The ID of the `Segment` onto which the join is made.
+  */
+  string segmentId;
+  /**
+  The 0-based position on the target `Segment` at which the join occurs.
+  */
+  int position;
+  /**
+  The face onto which the join is made. `PLUS` means that the join occurs on the
+  downstream face of the indicated base, while `MINUS` means that the join
+  occurs on the upstream face.
+  */
+  SegmentSide side;
+}
+
+/**
+A `Segment` represents a piece of sequence, which is optionally joined onto
+other `Segment`s of sequence at each end, forming a graph. The bases for a
+`Segment` are available through the `ListSegmentBases` API call.
+*/
+record Segment {
+  /**
+  The ID of this `Segment`. Unique within the repository.
+  */
+  string id;
+
+  /**
+  The length of this `Segment`'s sequence.
+  */
+  long length;
+
+  /**
+  Start and end locations of how this `Segment` fits into the existing graph.
+  May be null only for a primary sequence that is not joined onto any other.
+  Note that the segmentId for start and end may not be the same.
+  The `Segment`s onto which this `Segment` is joined are called its "parents".
+  */
+  union { null, SegmentJoinLocation } startJoin, endJoin;
+
+  /**
+  ID of the `VariantSet` that the `Segment` belongs to, if any.
+
+  If null, this `Segment` has an associated `Reference`.
+  */
+  union { null, string } variantSetId = null;
```

Comment:
By removing this field, we could have SNPs from two VariantSets (e.g. UK10K and 1000g) linked to the same `Segment`. This would seem nice. What backward dependency are you thinking of?